src/Aquifer.Jobs/Subscribers/UploadResourceContentAudioMessageSubscriber.cs

```diff
@@ -0,0 +1,421 @@
+        string mp3FilePath = null!;
+        int mp3FileSize;
+        string webmFilePath = null!;
+        int webmFileSize;
+        try
+        {
+            // download temp blob (keeping the same file name and extension)
+            tempAudioFilePath = GetTempFilePath(Path.GetFileName(message.TempUploadBlobName));
+            await _blobStorageService.DownloadFileAsync(
+                _uploadOptions.TempStorageContainerName,
+                message.TempUploadBlobName,
+                tempAudioFilePath,
+                ct);
+
+            // normalize audio file
+            normalizedAudioFilePath = await NormalizeAudioFileAsync(tempAudioFilePath, ct);
+
+            // compress to output formats
+            mp3FilePath = await CompressToMp3Async(normalizedAudioFilePath, ct);
+            mp3FileSize = (int)new FileInfo(mp3FilePath).Length;
+
+            webmFilePath = await CompressToWebmAsync(normalizedAudioFilePath, ct);
+            webmFileSize = (int)new FileInfo(webmFilePath).Length;
+
+            // upload files to CDN
+            await _cdnBlobStorageService.UploadFilesInParallelAsync(
+                _cdnOptions.AquiferContentContainerName,
+                [
+                    (string.Format(cdnBlobFormatString, "mp3"), mp3FilePath),
+                    (string.Format(cdnBlobFormatString, "webm"), webmFilePath),
+                ],
+                overwrite: true, // overwrite existing files in case this job has to be replayed
+                ct);
+        }
+        finally
+        {
+            // ensure we delete all the temp files
+            SafeDelete(tempAudioFilePath);
+            SafeDelete(normalizedAudioFilePath);
+            SafeDelete(mp3FilePath);
+            SafeDelete(webmFilePath);
+
+            static void SafeDelete(string? filePath)
+            {
+                if (filePath == null)
+                {
+                    return;
+                }
+
+                try
+                {
+                    if (File.Exists(filePath))
+                    {
+                        File.Delete(filePath);
+                    }
+                }
+                catch
+                {
+                    // ignore errors deleting temp files
+                }
+            }
+        }
+
+        var cdnUrlFormatString = $"{_cdnOptions.CdnBaseUri}{_cdnOptions.AquiferContentContainerName}/{cdnBlobFormatString}";
+
+        // Update ResourceContentVersion.Content with new CDN URLs.
+        var hasSteps = audioContent.Mp3?.Steps?.Any() == true && audioContent.Webm?.Steps?.Any() == true;
+        if (message.StepNumber.HasValue)
+        {
+            var mp3Step = audioContent.Mp3?.Steps?.FirstOrDefault(s => s.StepNumber == message.StepNumber);
+            var webmStep = audioContent.Webm?.Steps?.FirstOrDefault(s => s.StepNumber == message.StepNumber);
+
+            if (mp3Step == null || webmStep == null)
+            {
+                throw new InvalidOperationException(
```

Comment:
If these exceptions throw then we'll have uploaded a file to the CDN but the ResourceContentVersion.Content won't be referencing it.  I thought about refactoring but it got messier to split the validation and assignment.  The endpoint is already validating this so I think this is sufficient and should never happen in practice.